File: 350. Intersection of Two Arrays II.py
class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        res=[]
        i=0
        while i<len(nums1):
            if len(nums2)>0:
                if nums1[i] in nums2:
                    res+=[nums1[i]]
                    nums2.pop(nums2.index(nums1[i]))
                    nums1.pop(i)
                else:
                    i+=1
            else:
                break
        return res

File: 350. Intersection of Two Arrays II.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        vector <int> res;
        for (int i = 0; i < nums1.size(); i++)
        {
            if (nums2.size() > 0)
            {
                vector <int> :: const_iterator pos;
                pos = find (nums2.begin(), nums2.end(), nums1[i]);
                if (pos != nums2.end())
                {
                    res.push_back(nums1[i]);
                    nums2.erase(pos);
                }
            }
        }
        return res;
    }
};
